query/src/main/java/org/infinispan/query/clustered/commandworkers/CQLazyFetcher.java

```diff
@@ -31,7 +31,7 @@ Object extractValue(DocumentExtractor extractor, int docIndex) {
       }
       if (projection == null) {
          Object key = extractKey(extractor, docIndex);
-         return cache.get(key);
+         return cache.get(fromStorage(key));
```

Comment:
Here too, if the cache injected in components has request media type == storage media type, then the conversion isn't necessary.

Reply:
This cache has the default request type, but the QueryInterceptor now indexes keys in the storage format. I did that change because it was always indexing the key in the request format, which is inefficient, since the ```QueryInterceptor``` was already receiving the key in the storage format (and was converting again).

Reply:
I agree 100% that `QueryInterceptor` should index the key in the storage format. 

My assertion was that the `Cache` instance should *not* have the default request type, in fact it should not be an `EncoderCache` instance at all.

Reply:
let me see if I can make a non-encoded cache arrive to this command, can't make any promises

Reply:
ok, I changed it